packages/flutter_tools/lib/src/template.dart

```diff
@@ -202,8 +230,24 @@ class Template {
   }
 }
 
-Directory templateDirectoryInPackage(String name) {
-  final String templatesDir = globals.fs.path.join(Cache.flutterRoot,
+Directory templateDirectoryInPackage(String name, FileSystem fileSystem) {
+  final String templatesDir = fileSystem.path.join(Cache.flutterRoot,
       'packages', 'flutter_tools', 'templates');
-  return globals.fs.directory(globals.fs.path.join(templatesDir, name));
+  return fileSystem.directory(fileSystem.path.join(templatesDir, name));
+}
+
+// Returns the directory containing the 'name' template directory in
+// flutter_template_images, to resolve image placeholder against.
+Directory templateImageDirectory(String name, FileSystem fileSystem) {
+  final String packageFilePath = fileSystem.path.join(
```

Comment:
There might be cases where this .packages file doesn't exist. I think the zip uploaded SDKs do not include the packages since they would contains paths of the zipping machine they would be incorrect for clients.

Less commonly, the other way this could go wrong is that the user could update or delete their pub cache. Since the tool is snapshotted, it is still able to start even if the .packages is pointing to junk.

First you need to check for the existence of the .packages and the imagePackageLibDir. If either is missing you'll need to run a pub get in the flutter_tools directory. It might be easier to hoist this work out of template creation into a separate service class and drill that down instead

Reply:
Done, thanks for flagging that as I'd never have hit it in my testing.

I haven't done a full sweep of tests locally so I may need to adjust the environment in some other tests the way I've done in project_tests.